projects/ngx-maplibre-gl/src/lib/source/source.directive.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DestroyRef,
+  Directive,
+  OnInit,
+  inject,
+  input,
+  signal,
+} from '@angular/core';
+import { Subject, fromEvent } from 'rxjs';
+import { filter, switchMap, tap } from 'rxjs/operators';
+import { MapService } from '../map/map.service';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Source, SourceSpecification } from 'maplibre-gl';
+
+/**
+ * `mgl-canvas-source` - a canvas source component
+ * @see [canvas](https://maplibre.org/maplibre-style-spec/sources/#canvas)
+ *
+ * @category Source Components
+ */
+@Directive({
+  standalone: true,
+})
+export class SourceDirective implements OnInit {
+  /** Init injection */
+  readonly mapService = inject(MapService);
+  private readonly destroyRef = inject(DestroyRef);
+
+  /**  Init input */
+  readonly id = input.required<string>();
+  readonly sourceId = signal<string | null>(null);
+
+  private readonly loadSourceSubject = new Subject<void>();
+  readonly loadSource$ = this.loadSourceSubject.asObservable();
+
+  constructor() {
+    this.destroyRef.onDestroy(() => this.removeSource.bind(this));
```

Comment:
Do we really need `bind(this)`? Can't we simply use `() => this.removeSource()`

Reply:
yes, without "bind" its make an error in mapService

Reply:
Did you try it with the lambda?

Reply:
yes it doesn't work

Reply:
Can you think of a reason why this would create this error? It's very strange to me...